src/components/select/select-list/select-list.component.js

```diff
@@ -319,29 +327,35 @@ const SelectList = React.forwardRef(
     }, [assignListWidth]);
 
     useLayoutEffect(() => {
-      let newHeight;
+      if (isOpen) {
+        let newHeight;
 
-      newHeight = listRef.current.clientHeight;
+        newHeight = listRef.current.clientHeight;
 
-      if (listActionButtonRef.current) {
-        newHeight += listActionButtonRef.current.parentElement.clientHeight;
-      }
+        if (listActionButtonRef.current) {
+          newHeight += listActionButtonRef.current.parentElement.clientHeight;
+        }
 
-      setListHeight(`${newHeight}px`);
-    }, [children]);
+        setListHeight(`${newHeight}px`);
+      }
+    }, [children, isOpen]);
 
     useEffect(() => {
       const keyboardEvent = "keydown";
       const listElement = listRef.current;
 
-      window.addEventListener(keyboardEvent, handleGlobalKeydown);
-      listElement.addEventListener("scroll", handleListScroll);
+      if (isOpen) {
+        window.addEventListener(keyboardEvent, handleGlobalKeydown);
+        listElement.addEventListener("scroll", handleListScroll);
+      }
 
       return function cleanup() {
-        window.removeEventListener(keyboardEvent, handleGlobalKeydown);
-        listElement.removeEventListener("scroll", handleListScroll);
+        if (isOpen) {
+          window.removeEventListener(keyboardEvent, handleGlobalKeydown);
+          listElement.removeEventListener("scroll", handleListScroll);
+        }
       };
```

Comment:
the same issue as with scroll blocking above, the event listeners should be removed regardless of the `isOpen` state.